packages/core/src/virtual-test-entrypoint.ts

```diff
@@ -0,0 +1,201 @@
+      fromFile: m.groups!.filename,
+    };
+  }
+}
+
+export function renderTestEntrypoint(
+  resolver: Resolver,
+  { fromFile }: { fromFile: string }
+): { src: string; watches: string[] } {
+  const owner = resolver.packageCache.ownerOfFile(fromFile);
+
+  if (!owner) {
+    throw new Error('Owner expected');
+  }
+
+  let engine = resolver.owningEngine(owner);
+  let hasFastboot = Boolean(resolver.options.engines[0]!.activeAddons.find(a => a.name === 'ember-cli-fastboot'));
+
+  let appFiles = new AppFiles(
+    {
+      package: owner,
+      addons: new Map(
+        engine.activeAddons.map(addon => [
+          resolver.packageCache.get(addon.root) as V2AddonPackage,
+          addon.canResolveFromFile,
+        ])
+      ),
+      isApp: true,
+      modulePrefix: resolver.options.modulePrefix,
+      appRelativePath: 'NOT_USED_DELETE_ME',
+    },
+    getAppFiles(owner.root),
+    hasFastboot ? getFastbootFiles(owner.root) : new Set(),
+    extensionsPattern(resolver.options.resolvableExtensions),
+    staticAppPathsPattern(resolver.options.staticAppPaths),
+    resolver.options.podModulePrefix
+  );
+
+  let amdModules: { runtime: string; buildtime: string }[] = [];
+
+  for (let relativePath of appFiles.tests) {
+    amdModules.push(importPaths(resolver, appFiles, relativePath));
+  }
+
+  let src = entryTemplate({
+    amdModules,
+    testSuffix: true,
+    // this is a backward-compatibility feature: addons can force inclusion of test support modules.
+    defineModulesFrom: './-embroider-implicit-test-modules.js',
+  });
+
+  return {
+    src,
+    watches: [],
+  };
+}
+
+const entryTemplate = compile(`
+import { importSync as i, macroCondition, getGlobalConfig } from '@embroider/macros';
+let w = window;
+let d = w.define;
+
+{{#if styles}}
+  if (macroCondition(!getGlobalConfig().fastboot?.isRunning)) {
+    {{#each styles as |stylePath| ~}}
+      i("{{js-string-escape stylePath.path}}");
+    {{/each}}
+  }
+{{/if}}
+
+{{#if defineModulesFrom ~}}
+  import implicitModules from "{{js-string-escape defineModulesFrom}}";
+
+  for(const [name, module] of Object.entries(implicitModules)) {
+    d(name, function() { return module });
+  }
+{{/if}}
+
+
+import "ember-testing";
+import "@embroider/core/entrypoint";
+
+{{#each amdModules as |amdModule| ~}}
+  d("{{js-string-escape amdModule.runtime}}", function(){ return i("{{js-string-escape amdModule.buildtime}}");});
+{{/each}}
+
+{{#if fastbootOnlyAmdModules}}
+  if (macroCondition(getGlobalConfig().fastboot?.isRunning)) {
+    let fastbootModules = {};
+
+    {{#each fastbootOnlyAmdModules as |amdModule| ~}}
+      fastbootModules["{{js-string-escape amdModule.runtime}}"] = import("{{js-string-escape amdModule.buildtime}}");
+    {{/each}}
+
+    const resolvedValues = await Promise.all(Object.values(fastbootModules));
+
+    Object.keys(fastbootModules).forEach((k, i) => {
+      d(k, function(){ return resolvedValues[i];});
+    })
+  }
+{{/if}}
+
+
+{{#if lazyRoutes}}
```

Comment:
I think a test entrypoint would never use this.

Reply:
removed 👍